src/applications/bmqstoragetool/m_bmqstoragetool_compositesequencenumber.cpp

```diff
@@ -28,75 +28,84 @@ namespace m_bmqstoragetool {
-        return *this;  // RETURN
-    }
-
-    // Extract parts
-    const bsl::string firstPart  = seqNumString.substr(0, separatorPos);
-    const bsl::string secondPart = seqNumString.substr(separatorPos + 1);
-
-    // Convert parts to numbers
-    try {
-        size_t posFirst, posSecond;
-
-        unsigned long uLong = bsl::stoul(firstPart, &posFirst);
-        d_seqNumber         = bsl::stoul(secondPart, &posSecond);
+    // PRECONDITION
+    BSLS_ASSERT(success);
 
-        if (posFirst != firstPart.size() || posSecond != secondPart.size()) {
-            throw bsl::invalid_argument("");  // THROW
+    do {
+        if (seqNumString.empty()) {
+            errorDescription << "Invalid input: empty string.";
+            break;  // BREAK
         }
 
-        d_leaseId = static_cast<unsigned int>(uLong);
-        if (uLong != d_leaseId) {
-            throw bsl::out_of_range("");  // THROW
+        // Find the position of the separator
+        const size_t separatorPos = seqNumString.find('-');
+        if (separatorPos == bsl::string::npos) {
+            errorDescription << "Invalid format: no '-' separator found.";
+            break;  // BREAK
         }
 
-        if (d_leaseId == 0 || d_seqNumber == 0) {
-            errorDescription << "Invalid input: zero values encountered.";
+        // Extract parts
+        const bsl::string firstPart  = seqNumString.substr(0, separatorPos);
+        const bsl::string secondPart = seqNumString.substr(separatorPos + 1);
+
+        // Convert parts to numbers
+        try {
+            size_t posFirst, posSecond;
+
+            const unsigned long       uLong = bsl::stoul(firstPart, &posFirst);
+            const bsls::Types::Uint64 seqNumber = bsl::stoul(secondPart,
+                                                             &posSecond);
+            if (posFirst != firstPart.size() ||
+                posSecond != secondPart.size()) {
+                errorDescription
+                    << "Invalid input: non-numeric values encountered.";
+                break;  // BREAK
+            }
+
+            unsigned int leaseId = static_cast<unsigned int>(uLong);
+            if (uLong != leaseId) {
+                errorDescription << "Invalid input: number out of range.";
+                break;  // BREAK
+            }
+
+            if (leaseId == 0 || seqNumber == 0) {
+                errorDescription << "Invalid input: zero values encountered.";
+                break;  // BREAK
+            }
+
+            d_compositeSequenceNumber = bsl::make_pair(leaseId, seqNumber);
+
+            *success = true;
             return *this;  // RETURN
         }
+        catch (const bsl::invalid_argument& e) {
+            errorDescription
+                << "Invalid input: non-numeric values encountered.";
+        }
+        catch (const bsl::out_of_range& e) {
+            errorDescription << "Invalid input: number out of range.";
+        }
```

Comment:
What about placing `try-catch` block around the minimal code that might produce exceptions? In this case, only `bsl::stoul` lines should be covered